smpclient/transport/serial.py

```diff
@@ -78,7 +78,10 @@ async def disconnect(self) -> None:
 
     async def send(self, data: bytes) -> None:
         logger.debug(f"Sending {len(data)} bytes")
-        for packet in smppacket.encode(data, line_length=self.mtu):
+        # line cannot be bigger than 128 bytes by default
+        # https://github.com/zephyrproject-rtos/zephyr/blob/076c0df3d8a878e4457c1d831704cd33b9ead6d3/drivers/console/uart_mcumgr.c#L101
+        # https://github.com/zephyrproject-rtos/zephyr/blob/076c0df3d8a878e4457c1d831704cd33b9ead6d3/drivers/console/Kconfig#L204
+        for packet in smppacket.encode(data, line_length=128):
```

Comment:
Which soc are you testing? Nrf-sdk with nrf52840 won't work that way. Single line must be 128 and in the same time mtu has to be bigger than that. Otherwise there is parsing error in mcumgr at zephyr side.

Reply:
I've tested on STM32 and NRF52840.  However, in both cases I have only used the SMP server provided in MCUBoot.  In case of NRF52840 the system has serial DFU via MCUBoot and BLE DFU via the application.